AppKit/CPTrackingArea.j

```diff
@@ -0,0 +1,143 @@
+    CPTrackingAreaOwnerKey           = @"CPTrackingAreaOwnerKey",
+    CPTrackingAreaUserInfoKey        = @"CPTrackingAreaUserInfoKey",
+    CPTrackingAreaReferencingViewKey = @"CPTrackingAreaReferencingViewKey",
+    CPTrackingAreaActualRect         = @"CPTrackingAreaActualRect";
+
+
+/*!
+ @ingroup appkit
+ 
+ A CPTrackingArea defines a region of view that generates mouse-tracking and
+ cursor-update events when the mouse is over that region.
+ */
+@implementation CPTrackingArea : CPObject
+{
+    CGRect                  _rect               @accessors(getter=rect);
+    CPTrackingAreaOptions   _options            @accessors(getter=options);
+    id                      _owner              @accessors(getter=owner);
+    CPDictionary            _userInfo           @accessors(getter=userInfo);
+    
+    CPView                  _referencingView    @accessors;
+    CGRect                  _actualRect         @accessors(getter=actualRect);
+}
+
+
+#pragma mark -
+#pragma mark Initialization
+
+/*! Initializes and returns an object defining a region of a view to receive mouse-tracking events, mouse-moved events, cursor-update events, or possibly all these events.
+ */
+- (CPTrackingArea)initWithRect:(CGRect)aRect options:(CPTrackingAreaOptions)options owner:(id)owner userInfo:(CPDictionary)userInfo
+{
+    if (self = [super init])
+    {
+        if (options === 0)
+            [CPException raise:CPInternalInconsistencyException reason:"Invalid CPTrackingArea options"];
+
+        // Check options:
+        // - at least one of CPTrackingMouseEnteredAndExited, CPTrackingMouseMoved, CPTrackingCursorUpdate
+        // - exactly  one of CPTrackingActiveWhenFirstResponder, CPTrackingActiveInKeyWindow, CPTrackingActiveInActiveApp, CPTrackingActiveAlways
+        // - no check on CPTrackingAssumeInside, CPTrackingInVisibleRect, CPTrackingEnableDuringMouseDrag
+        
+        if (!((options & CPTrackingMouseEnteredAndExited) || (options & CPTrackingMouseMoved) || (options & CPTrackingCursorUpdate)))
+            [CPException raise:CPInternalInconsistencyException reason:"Invalid CPTrackingAreaOptions: must use at least one of [CPTrackingMouseEnteredAndExited | CPTrackingMouseMoved | CPTrackingCursorUpdate]"];
+        
+        if ((((options & CPTrackingActiveWhenFirstResponder) > 0) + ((options & CPTrackingActiveInKeyWindow) > 0) + ((options & CPTrackingActiveInActiveApp) > 0) + ((options & CPTrackingActiveAlways) > 0)) !== 1)
+            [CPException raise:CPInternalInconsistencyException reason:"Tracking area options may only specify one of [CPTrackingActiveWhenFirstResponder | CPTrackingActiveInKeyWindow | CPTrackingActiveInActiveApp | CPTrackingActiveAlways]."];
+
+        _rect     = aRect;
+        _options  = options;
+        _owner    = owner;
+        _userInfo = userInfo;
+    }
+    
+    return self;
+}
+
+
+#pragma mark -
+#pragma mark Implementation
+
+- (BOOL)_isReferenced
```

Comment:
Please remove this method and replace the only place it is called with `if ([trackingArea view])`